### PR TITLE
Vacancies

### PR DESCRIPTION
**Pull request by g4s8**
#925 - add vacancies to `vacancies.xml` in `TkHiring`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#953](https://github.com/zerocracy/farm/issues/953) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#953](https://github.com/zerocracy/farm/issues/953) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1101046010); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/com/zerocracy/pmo/Vacancies.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import org.cactoos.time.DateAsText;
+import org.xembly.Directives;
+
+/**
+ * Available vacancies.
+ *
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id $
+ * @since 0.22
+ * @todo #925:30min Vacancies are not rendered on `/vacancies` page.
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pmo/Vacancies.java]: @g4s8 let's separate these into two tasks, they seem distinct to me. One task to render `/vacancies` data and another to clean up old vacancies.

File: src/test/java/com/zerocracy/pmo/VacanciesTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+public final class VacanciesTest {
+    @Test
+    public void addNewVacancy() throws Exception {
+        final Vacancies vacancies = new Vacancies(new FkFarm()).bootstrap();
+        final Project project = new FkProject();
+        vacancies.add(project, "test", "unit test is hiring");
+        MatcherAssert.assertThat(
+            vacancies.iterate(),
+            Matchers.allOf(
+                Matchers.iterableWithSize(1),
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/VacanciesTest.java]: @g4s8 we can simplify this by using [Matchers.contains](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#contains(org.hamcrest.Matcher)), which is a special case that matches exactly one item corresponding to the given `Matcher`.

File: src/test/java/com/zerocracy/pmo/VacanciesTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Test case for {@link Vacancies}.
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id$
+ * @since 0.22
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class VacanciesTest {
+    @Test
+    public void addNewVacancy() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/VacanciesTest.java]: @g4s8 let's name this `addsVacancy`.

File: src/test/java/com/zerocracy/pmo/VacanciesTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            vacancies.iterate(),
+            Matchers.allOf(
+                Matchers.iterableWithSize(1),
+                Matchers.hasItem(Matchers.equalTo(project.pid()))
+            )
+        );
+    }
+
+    @Test
+    public void removeVacancy() throws Exception {
```
Comment (carlosmiranda) [inline src/test/java/com/zerocracy/pmo/VacanciesTest.java]: @g4s8 let's call this `removesVacancy`

**Comment by g4s8** [comment]
@carlosmiranda fixed in f632cca

File: src/main/java/com/zerocracy/pmo/Vacancies.java (hunk 1 above)
Comment (adouzhinan) [reply src/main/java/com/zerocracy/pmo/Vacancies.java]: can

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#953](https://github.com/zerocracy/farm/issues/953) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@adouzhinan According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please provide an addressee in the beginning of your [code review comment](https://github.com/zerocracy/farm/pull/953#discussion_r183296902).

**Comment by carlosmiranda** [comment]
@ypshenychka @yegor256 can we ignore the message from @adouzhinan . It appears to be some sort of throwaway account, he joined just last week and posted no activity other than those messages on this ticket.

**Comment by yegor256** [comment]
@ypshenychka yes, we can

**Comment by ypshenychka** [comment]
@yegor256 thanks

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**